### ISSUE TITLE
salt/states/test.py check_pillar() present option does not work

### ISSUE THREAD

**Issue by kevinquinnyo** [saltstack/salt#32964]
### Description of Issue/Question
Using the `present` keyword will never work because the private function `_check_key_type()` receives `None` in that case.  As such it will always return false.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
Easy to test:
```
#!py

def run():
    states = {}
    states['Test test check_pillar with present option'] = {
        'test.check_pillar': [
            {'present': 'foo'}
        ] 
    }
    return states
```

Run: `# salt 'some-target' state.sls test pillar="{'foo': 'bar'}"`

### Steps to Reproduce Issue
Not applicable, see above.

### Versions Report
This is present in the develop branch and probably all versions that include the state, but I'm running:
salt-master 2015.8.4 (Beryllium)
salt-minion 2015.8.7 (Beryllium)

**Comment by jfindlay** [comment]
@kevinquinnyo, thanks for reporting.

**Comment by rallytime** [comment]
Fixed by #32965
